### PR TITLE
Glob except

### PR DESCRIPTION
**Pull request by AGuliaiev**

### CODE REVIEW

File: src/main/java/org/example/springintro/services/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                );
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(
+                        () -> new EntityNotFoundException("Book not found by id: " + id)
+                );
```
Comment (okuzan) [inline src/main/java/org/example/springintro/services/impl/BookServiceImpl.java]: extract repeated logic to a private method

File: src/main/java/org/example/springintro/services/BookService.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    List<BookDto> findAll();
+
+    BookDto findById(Long id);
+
+    BookDto updateById(Long id, CreateBookRequestDto requestDto);
+
+    void deleteById(Long id);
+
+    List<BookDto> search(BookSearchParameters params);
+
```
Comment (okuzan) [inline src/main/java/org/example/springintro/services/BookService.java]:
```suggestion
```

File: src/main/java/org/example/springintro/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Setter
+@Getter
+@SQLDelete(sql = "UPDATE books SET is_delete = true WHERE id=?")
+@SQLRestriction(value = "is_delete=false")
```
Comment (okuzan) [inline src/main/java/org/example/springintro/model/Book.java]:
```suggestion
@SQLDelete(sql = "UPDATE books SET is_delete = TRUE WHERE id = ?")
@SQLRestriction(value = "is_delete = FALSE")
```

File: pom.xml [hunk 4]
```diff
@@ -76,7 +96,67 @@
...
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
         </plugins>
+
     </build>
 
-</project>
+</project>
```
Comment (okuzan) [inline pom.xml]:
```suggestion
</project>

```

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Changes in PR are related not only to Global Exc HW, but to previous HWs also. 
You have 5 open PRs. If they were approved by mentors, merge to main and rebase this branch

**Pull request merged**